File: website/scripts/Block.js
//test image
const input = new Image();
const texture = new Image();

class Block{
  constructor(source, blocksize){
    this.src = source;
    this.startW = Math.random * source.width - blocksize;
    this.startH = Math.random * source.height - blocksize;
    this.endW = Math.min(startW + blockSize, source.width);
    this.endH = Math.min(startH + blockSize, source.height);
  }

  getBlock(){
    
  }
}

input.src = 'https://static.wikia.nocookie.net/dragons/images/e/ed/Chinese-dragon-red.jpg/';
texture.src = 'https://i.pinimg.com/736x/64/14/98/641498d8c60f30f2d4b725dce69c6bfd.jpg';

File: spring/TextureTransfer/src/main/java/com/texture/TextureTransfer/Block.java
package com.texture.TextureTransfer;

import java.awt.image.BufferedImage;

public class Block {
    int endW;
    int endH;
    int startW;
    int startH;

    BufferedImage src;

    public Block(BufferedImage src, int blockSize) {
        this.src = src;
        startW = (int) (Math.random() * (src.getWidth() - blockSize));
        startH = (int) (Math.random() * (src.getHeight() - blockSize));
        endW = Math.min(startW + blockSize, src.getWidth());
        endH = Math.min(startH + blockSize, src.getHeight());
    }

    public BufferedImage getBlock() {
        BufferedImage result = new BufferedImage(endW - startW, endH - startH, src.getType());

        for (int i = 0; i < endW - startW; i++) {
            for (int j = 0; j < endH - startH; j++) {
                int rgb = src.getRGB(i + startW, j + startH);
                result.setRGB(i, j, rgb);
            }
        }

        return result;
    }
}
